core/templates/dev/head/pages/creator-dashboard-page/creator-dashboard-page.controller.ts

```diff
@@ -247,10 +249,14 @@ oppia.directive('creatorDashboardPage', ['UrlInterpolationService', function(
             ctrl.lastWeekStats = responseData.last_week_stats;
             ctrl.myExplorationsView = responseData.display_preference;
             ctrl.topicSummaries = responseData.topic_summary_dicts;
-            var numberOfCreatedSuggestions = (
-              responseData.threads_for_created_suggestions_list.length);
-            var numberOfSuggestionsToReview = (
-              responseData.threads_for_suggestions_to_review_list.length);
+            if (responseData.threads_for_created_suggestions_list) {
```

Comment:
This change seems odd. If the condition doesn't hold then numberOfCreatedSuggestions will be undefined and the for loop on line 261 will not run correctly. Why make this change in this PR?

Reply:
That change was part of the debugging during Karma failures. This was giving undefined value in the spec so its length couldn't be read properly. 
Also, in general we assume here that the `responseData.threads_for_created_suggestions_list` is having something (because otherwise `responseData.threads_for_suggestions_to_review_list.length` would throw an exception), so having this condition shouldn't hurt I guess?

Reply:
If you are assuming that the value exists then the "if" condition should not be needed.

Is the spec incorrect? You would need to find why the value is undefined, maybe the response in the spec is defined wrongly. But this is not the fix.

Reply:
Ok, understood.

Also, that fix isn't part of this PR, that was something I tried when I was modularizing. Therefore, I have discarded these changes.